client/signup/config/flows.js

```diff
@@ -328,16 +335,15 @@ const Flows = {
 		}
 
 		/**
-		 * This is called on Signup start (page initialization),
-		 * before the steps are rendered. Used if there is a need
-		 * to filter the first step in the flow.
+		 * If there is need to test the first step in a flow,
+		 * the best way to do it is to check for:
+		 *
+		 * 	if ( '' === stepName ) { ... }
+		 *
+		 * This will be fired at the beginning of the signup flow.
 		 */
-		if ( '' === stepName ) {
-			abtest( 'noSurveyStep' );
-		}
-
 		if ( 'main' === flowName ) {
-			if ( 'survey' === stepName ) {
+			if ( '' === stepName ) {
```

Comment:
I don't know that it would have any effect, since the AB would already be set, but is it possible to have an empty string as the stepName right after Signup is complete (before proceeding to checkout)?